File: js/remove_element.js
//https://leetcode.com/problems/remove-element/description/

/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {

    var i = 0;
    while (i<nums.length){

        if(nums[i] == val){
            temp = nums.pop()
            if (nums.length == i){
                return nums.length
            };
            nums[i] = temp;

        }else{
            i += 1
        };

    }
    return nums.length;

};

File: leetcode/python/remove_element.py
# https://leetcode.com/problems/remove-element/description/

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i = 0
        while i < len(nums):
            if nums[i] == val:
                temp = nums.pop()
                if len(nums) == i:
                    return len(nums)
                nums[i] = temp
            else:
                i += 1
        return len(nums)